app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java

```diff
@@ -150,6 +155,44 @@ public void addEventToCalendar(SimpleEvent event) {
         startActivity(intent);
     }
 
+    /**
+     * Helper method that will split events into upcoming / past
+     * and add to the adapter of this list fragment
+     *
+     * @param events
+     *      The events list to be added
+     */
+    protected void splitEventsAndAddToAdapter(Collection<? extends SimpleEvent> events) {
+        Pair<List<SimpleEvent>, List<SimpleEvent>> pair = splitEventsList(events);
+        mAdapter.addAll(pair.first);
+        mAdapter.addAll(pair.second);
+    }
+
+    /**
+     * Split a events list into Upcoming / Past events
+     *
+     * @param events
+     *      The events list that needs to be sorted
+     * @return
+     *      A pair containting the split list into Upcoming / Past events
+     */
+    protected Pair<List<SimpleEvent>, List<SimpleEvent>> splitEventsList(Collection<? extends SimpleEvent> events) {
+        List<SimpleEvent> upcoming = new ArrayList<>();
+        List<SimpleEvent> past = new ArrayList<>();
+
+        DateTime now = DateTime.now();
+
+        for (SimpleEvent event : events) {
+            if (event.getStart().isBefore(now)) {
+                past.add(event);
+            } else {
+                upcoming.add(event);
+            }
+        }
+        Collections.reverse(past);

```

Comment:
No. This actually sorts like this. The Upcoming event list is sorted in ASC order as we want the ones that are the closest to the date we are first and the Past events order is sorted in DESC order as we want the most recents one on top.